OpenRA.Game/Traits/TraitsInterfaces.cs

```diff
@@ -96,14 +96,15 @@ public Damage(int damage)
 	public interface ITick { void Tick(Actor self); }
 	[RequireExplicitImplementation]
 	public interface ITickRender { void TickRender(WorldRenderer wr, Actor self); }
-	public interface IRender { IEnumerable<IRenderable> Render(Actor self, WorldRenderer wr); }
+	public interface IRender
+	{
+		IEnumerable<IRenderable> Render(Actor self, WorldRenderer wr);
+		Rectangle AutoRenderBounds(Actor self);
```

Comment:
I think this would be best defined as `IEnumerable<Rectangle> ScreenBounds(Actor self, WorldRenderer wr)`, and then have the implementing traits return the screen-space rectangles of each animation.  This then avoids N different places having to aggregate over the rectangles that they "own".